error/s2n_errno.h

```diff
@@ -282,6 +282,7 @@ typedef enum {
     S2N_ERR_INVALID_X509_EXTENSION_TYPE,
     S2N_ERR_INSUFFICIENT_MEM_SIZE,
     S2N_ERR_KEYING_MATERIAL_EXPIRED,
+    S2N_ERR_EXTENSION_NOT_FOUND,
     S2N_ERR_T_USAGE_END,
```

Comment:
Are you sure S2N_ERR_INVALID_ARGUMENT isn't specific enough? (Not disagreeing, just want to hear the reasons)

Reply:
Searched and it seems S2N_ERR_INVALID_ARGUMENT is widely used for the case when could not find for request data.